src/adios4dolfinx/adios2_helpers.py

```diff
@@ -3,7 +3,12 @@
 
 from mpi4py import MPI
 
+import packaging.version
 import adios2
+if (packaging.version.parse(adios2.__version__)
+        >= packaging.version.Version("2.10.0")):
+    adios2 = adios2.bindings
+
```

Comment:
Could we use: 
https://docs.python.org/3/library/importlib.metadata.html
for the versioning?

Reply:
Or something like
```
try:
    import adios2.bindings as adios2
except ImportError:
    import adios2
```
?

Reply:
Yep, this might be better in case they revert the bindings at some point

Reply:
The disadvantage with the try catch is the yielded exception may be confusing regarding the true nature of the problem. I.e. does `adios2` exist, or is it simply a scope resolution issue between versions of the package.

Reply:
what about:
```python
def import_adios():
    try:
        import adios2
        return adios2.bindings
    except AttributeError:
        import adios2
        return adios2
````